### PR TITLE
add reference to spack help --spec to spack spec -h

### PR DESCRIPTION
**Pull request by healther**
@adamjstewart mentioned this offhand in some other PR or issue, but I had to look it up again today, so mostly convenience for me (and other lazy/forgetful people)

### CODE REVIEW

**Comment by adamjstewart** [comment]
I agree that it is useful to remind people of this help flag (I'm glad I convinced @tgamblin not to remove it). However, although this looks good for `spack help spec`, the same `description` is used for `spack help`, where it doesn't look so good. Maybe we need some kind of `long_description` specific to `spack help <subcommand>`? I don't know how this would work, @tgamblin is more familiar with our argparse hacks.

The other thing I'm worried about is that this spec syntax isn't specific to `spack spec`, it is used for almost every Spack command. That's why it is currently in the generic `spack help` and not already a part of `spack help spec`. @tgamblin would it be overkill to add this text to every subcommand that accepts the spec syntax?

**Comment by healther** [comment]
I'll take a stab at removing it from the general help, but if that turns out to be complicated I'd be for merging this with minor text changes to make it less painful to see in `spack help`, as we really shouldn't make it hard to find this

**Comment by healther** [comment]
This now gives:
```
$ spack help
usage: spack [-hkV] [--color {always,never,auto}] COMMAND ...

A flexible package manager that supports multiple versions,
configurations, platforms, and compilers.

These are common spack commands:

query packages:
  list                  list and search available packages
  info                  get detailed information on a particular package
  find                  list and search installed packages

build packages:
  install               build and install packages
  uninstall             remove installed packages
  spec                  show what would be installed, given a spec

environment:
  load                  add package to environment using `module load`
  module                manipulate module files
  unload                remove package from environment using `module unload`
  view                  produce a single-rooted directory view of packages

create packages:
  create                create a new package file
  edit                  open package files in $EDITOR

system:
  arch                  print architecture information about this machine
  compilers             list available compilers

optional arguments:
  -h, --help            show this help message and exit
  -k, --insecure        do not check ssl certificates when downloading
  -V, --version         show version number and exit
  --color {always,never,auto}
                        when to colorize output (default: auto)

more help:
  spack help --all       list all available commands
  spack help <command>   help on a specific command
  spack help --spec      help on the spec syntax
  spack docs             open http://spack.rtfd.io/ in a browser
```
and
```
$ spack spec -h
usage: spack spec [-hlLyNIt] [-c {nodes,edges,paths}] ...

show what would be installed, given a spec

For further documentation regarding the spec syntax see
    spack help --spec

positional arguments:
  specs                 specs of packages

optional arguments:
  -h, --help            show this help message and exit
  -l, --long            show dependency hashes as well as versions
  -L, --very-long       show full dependency hashes as well as versions
  -y, --yaml            print concrete spec as YAML
  -c {nodes,edges,paths}, --cover {nodes,edges,paths}
                        how extensively to traverse the DAG (default: nodes)
  -N, --namespaces      show fully qualified package names
  -I, --install-status  show install status of packages. packages can be: installed [+], missing and needed by an installed package [-], or not installed (no annotation)
  -t, --types           show dependency types
```

Obviously this needs to be cleaned up. Let me know what you think. Should I implement a second member for all `cmd` modules or should `main.py` be responsible for checking if a "long description" exists?

**Comment by healther** [comment]
ping

**Comment by healther** [comment]
@adamjstewart does this address your remarks?

File: lib/spack/spack/cmd/spec.py [hunk 1]
```diff
@@ -29,7 +29,13 @@
 import spack.cmd
 import spack.cmd.common.arguments as arguments
 
-description = "show what would be installed, given a spec"
+description = """show what would be installed, given a spec"""
+
+help = """
```
Comment (adamjstewart) [inline lib/spack/spack/cmd/spec.py]: I don't understand how this works. Shouldn't this variable be called `longdescription`?

**Comment by healther** [comment]
ping, do we want to merge this, if yes what should be done for it? or should I close this?

**Comment by tgamblin** [comment]
I like it but I have the same question as @adamjstewart above -- what's `longdescription` vs. `help` here?

**Comment by alalazo** [comment]
@healther This has been stale for a long time. Would it be fine to close it? Or do you want to revisit it later?

**Comment by healther** [comment]
@alalazo fine with me, I didn't stumble over this in the last year :)

**Pull request closed without merging**